### DISCUSSION TITLE
Custom path for items in listing

### DISCUSSION THREAD

**Discussion by tim-at-topos** [quarto-dev/quarto-cli#8130] [Q&A]
### Description

My Quarto project has the following structure:

```
quarto_project/
├─ blog/
│  ├─ 2022/
│  │  ├─ 2022-MM-DD-slug/
│  │  │  ├─ index.qmd
│  ├─ 2023/
│  │  ├─ 2023-MM-DD-slug/
│  │  │  ├─ index.qmd
│  ├─ index.qmd
```

In other words, each blog post is an `index.qmd` placed inside the directory `YYYY-MM-DD-slug/`, but to keep things easy to browse (since there are a lot of blog posts), these directories are placed inside a corresponding `YYYY/` directory.

The file `/blog/index.qmd` is a listing, with front matter

```
---
title: "Blog"
listing: 
  contents: .
  type: default
---
```

Which works almost perfectly, but the blog posts in the rendered site obviously have URLs of the form `/blog/YYYY/YYYY-MM-DD-slug`.

**How can I get Quarto to render these pages such that their URLs are of the form `/blog/YYYY-MM-DD-slug/`?**

**Comment by mcanouil** [comment]
This was discussed several times in discussions and issues (non exhaustive list):
- https://github.com/quarto-dev/quarto-cli/discussions/6418
- https://github.com/quarto-dev/quarto-cli/discussions/6398
- https://github.com/quarto-dev/quarto-cli/discussions/7299
- https://github.com/quarto-dev/quarto-cli/discussions/1408

**Reply by tim-at-topos** [reply]
So the method as of now is to do this manually with redirects in each file? Ok, thanks! I'll probably set up some pre-processing script to do just that then
